Essentials/src/com/earth2me/essentials/commands/Commandtppos.java

```diff
@@ -55,7 +56,7 @@ public void run(final Server server, final User user, final String commandLabel,
         final Trade charge = new Trade(this.getName(), ess);
         charge.isAffordableFor(user);
         user.sendMessage(tl("teleporting", loc.getWorld().getName(), loc.getBlockX(), loc.getBlockY(), loc.getBlockZ()));
-        user.getTeleport().teleport(loc, charge, TeleportCause.COMMAND);
+        user.getTeleport().teleport(loc, charge, TeleportCause.COMMAND, !NumberUtil.isInt(x) || !NumberUtil.isInt(z));
```

Comment:
This really isn't a safe clause here (while I understand what you are going for).

This is a problem in floating point comparisons - what is happening here is that x (a double) is being coerced into being a string for NumberUtil to then use Integer.parseInt() on underneath.

It would be a lot nicer (and more efficient) if this method had some way of instead detecting whether or not the user intended to simply enter an integer or whether they intended to enter a double.

I think an immediate bug would be the immediate one you are trying to work around - pretend for a moment that the user enters /tppos 42.0 42.0 42.0. The correct behavior would be to teleport the player directly to 42, 42, 42, not 42.5 42 42.5 like this code will actually do. 42.0 is an Integer, and will pass the checks presented.

Reply:
EDIT: I didn't see the overload on the bottom. My point is still valid for the error point I present.